src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1259,7 +1260,7 @@ abstract class RefChecks extends Transform {
           reporter.warning(pos, s"${sym.fullLocationString} has changed semantics in version ${sym.migrationVersion.get}:\n${sym.migrationMessage.get}")
       }
       // See an explanation of compileTimeOnly in its scaladoc at scala.annotation.compileTimeOnly.
-      if (sym.isCompileTimeOnly && !currentOwner.ownerChain.exists(x => x.isCompileTimeOnly)) {
+      if (sym.isCompileTimeOnly && !inAnnotation && sym != SpecializedClass && !currentOwner.ownerChain.exists(x => x.isCompileTimeOnly)) {
```

Comment:
Why add a `@compileTimeOnly` to `specialized` only to exclude it here?

Reply:
Ah, I just got the email for your comment on the other PR. Maybe throw types to the wind and say

```scala
lazy val SpecializedClass           = getRequiredClass("scala.specialized")
```

Reply:
Wait, what comment on the other PR? Which other PR? What is this PR about again?